src/main/java/todolist/service/impl/ToDoServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package todolist.service.impl;
+
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.AllArgsConstructor;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Service;
+import todolist.entity.ToDo;
+import todolist.repository.ToDoRepository;
+import todolist.service.ToDoService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class ToDoServiceImpl implements ToDoService {
+
+    private ToDoRepository repository;
+
+
+    @Override
+    public List<ToDo> findAll() {
+        return repository.findAll();
+    }
+
+
+    @Override
+    public ToDo findOne(Integer id) {
+        return repository.findById(id).orElseThrow(() -> new ResourceNotFoundException("ToDo not found with id: " + id));
+    }
+
+
+    @Override
+    public ToDo update(ToDo toDoRequest, Integer id) {
+        ToDo newToDo = repository.findById(id).orElseThrow(() -> new ResourceNotFoundException("ToDo not found with id: " + id));
+
+        newToDo.setTitle(toDoRequest.getTitle());
+        newToDo.setDueDate(toDoRequest.getDueDate());
+        newToDo.setStatus(toDoRequest.getStatus());
+
+        return repository.save(newToDo);
+    }
+
+
+    @Override
+    public ToDo create(ToDo toDoRequest) {
+        return repository.save(toDoRequest);
```

Comment:
What happens if save fails? Let's say a constraint on an attribute is not satisfied, like `unique`

Reply:
The ToDoExceptionHandler in the exceptionhandler package will take care of all validations and constraints before calling save() method